core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -1283,6 +1283,116 @@ public IRubyObject subseq(long beg, long len) {
         return subseq(getRuntime().getArray(), beg, len, true);
     }
 
+    /** rb_ary_subseq_step
+     *
+     */
+    public IRubyObject subseq_step(RubyArithmeticSequence arg0) {
+        long beg, len, step, end;
+
+        step = getStep(arg0);
+        IRubyObject aseqBeg = getBegin(arg0);
+        IRubyObject aseqEnd = getEnd(arg0);
+        boolean aseqExcl = arg0.exclude_end(getRuntime().getCurrentContext()).isTrue();
+        len = realLength;
+
+        if (step < 0) {
+            IRubyObject tmp = aseqBeg;
+            aseqBeg = aseqEnd;
+            aseqEnd = tmp;
+        }
+
+        if (aseqBeg.isNil()) {
+            beg = 0;
+        } else {
+            beg = aseqBeg.convertToFloat().getLongValue();
+        }
+
+        if (aseqEnd.isNil()) {
+            end = -1;
+        } else {
+            end = aseqEnd.convertToFloat().getLongValue();
+        }
+        if (aseqEnd.isNil()) {  aseqExcl = false; }
+
+        if (beg < 0) {
+            beg += len;
+            if (beg < 0) {
+                throw getRuntime().newRangeError("integer " + beg + " out of range of fixnum");
+            }
+        }
+        if (end < 0) {  end += len; }
+
+        if (!aseqExcl) {  end++; }
+
+        len = end - beg;
+
+        if (len < 0) {  len = 0;  }
+
+        long alen = realLength;
+        if (beg > alen)  return null;
```

Comment:
Some null returns here.  I think these should be nil?

Reply:
done